lib/ansible/modules/cloud/smartos/nictagadm.py

```diff
@@ -0,0 +1,238 @@
+
+    def __init__(self, module):
+        self.module = module
+
+        self.name = module.params['name']
+        self.mac = module.params['mac']
+        self.etherstub = module.params['etherstub']
+        self.mtu = module.params['mtu']
+        self.force = module.params['force']
+        self.state = module.params['state']
+
+    def is_valid_mac(self):
+        if re.match("[0-9a-f]{2}([:])[0-9a-f]{2}(\\1[0-9a-f]{2}){4}$", self.mac.lower()):
+            return True
+        else:
+            return False
+
+    def nictag_exists(self):
+        cmd = [self.module.get_bin_path('nictagadm', True)]
+
+        cmd.append('exists')
+        cmd.append(self.name)
+
+        (rc, dummy, dummy) = self.module.run_command(cmd)
+
+        if rc == 0:
+            return True
+        else:
+            return False
+
+    def add_nictag(self):
+        cmd = [self.module.get_bin_path('nictagadm', True)]
+
+        cmd.append('-v')
+        cmd.append('add')
+
+        if self.etherstub:
+            cmd.append('-l')
+
+        if self.mtu:
+            cmd.append('-p')
+            cmd.append('mtu=' + str(self.mtu))
+
+        if self.mac:
+            cmd.append('-p')
+            cmd.append('mac=' + str(self.mac))
+
+        cmd.append(self.name)
+
+        return self.module.run_command(cmd)
+
+    def delete_nictag(self):
+        cmd = [self.module.get_bin_path('nictagadm', True)]
+
+        cmd.append('-v')
+        cmd.append('delete')
+
+        if self.force:
+            cmd.append('-f')
+
+        cmd.append(self.name)
+
+        return self.module.run_command(cmd)
+
+
+def main():
+    module = AnsibleModule(
+        argument_spec=dict(
+            name=dict(required=True, type='str'),
+            mac=dict(default=None, type='str'),
+            etherstub=dict(default=False, type='bool'),
+            mtu=dict(default=None, type='int'),
+            force=dict(default=False, type='bool'),
+            state=dict(default='present', choices=['absent', 'present']),
+        ),
+        mutually_exclusive=[
+            ['etherstub', 'mac'],
+            ['etherstub', 'mtu'],
+        ],
+        required_if=[
+            ['etherstub', False, ['name', 'mac']],
+            ['state', 'absent', ['name', 'force']],
+        ],
+        supports_check_mode=True
+    )
+
+    nictag = NICTAG(module)
+
+    rc = None
+    out = ''
+    err = ''
+    result = {}
+    result['name'] = nictag.name
+    result['mac'] = nictag.mac
+    result['etherstub'] = nictag.etherstub
+    result['mtu'] = nictag.mtu
+    result['force'] = nictag.force
+    result['state'] = nictag.state
+
+    if not nictag.is_valid_mac():
+        module.fail_json(msg='Invalid MAC Address Value',
+                         name=nictag.name,
+                         mac=nictag.mac,
+                         etherstub=nictag.etherstub,
+                         mtu=nictag.mtu,
+                         force=nictag.force,
+                         state=nictag.state)
```

Comment:
Isn't this a bit too much context to report merely the MAC access is invalid? I'd include the `name`, `mac` and `etherstub` and leave out the others.